### DISCUSSION TITLE
StochasticTools; report input/sampled values sent to sub-app

### DISCUSSION THREAD

**Discussion by jvwilliams23** [idaholab/moose#27523] [Q&A Modules: General] [answered]
Hi,

I am using `StochasticTools` module for a monte carlo run as part of a UQ workflow. Using the `StochasticReporter` returns the final QoIs in the json file, but I am wondering if it is possible to also save the input parameters from the sampler which are sent to the sub-app to produce each QoI?

For example, the reporter and transfer
```
[Reporters]
  [storage]
    type = StochasticReporter
    parallel_type = ROOT
  []
[]

[Transfers]
  [sub]
    type = SamplerParameterTransfer
    to_multi_app = sub
    sampler = train_sampler_a
    parameters = 'Functions/slope_tc/value'
    check_multiapp_execute_on = false
  []
  [data]
    type = SamplerReporterTransfer
    from_multi_app = sub
    sampler = train_sampler_a
    stochastic_reporter = storage
    from_reporter = 'temp_xmin_wall/value'
  []
[]
```

yields the following output:
```
    "time_steps": [
        {
            "storage": {
                "data:converged": [
                    true,
                    true,
                    true,
                    true,
                    true,
                    true,
                    true,
                    true,
                    true,
                    true
                ],
                "data:temp_xmin_wall:value": [
                    327.21460091191886,
                    327.22095318699365,
                    327.2186406333933,
                    327.2131752782587,
                    327.2159796106987,
                    327.2169546408685,
                    327.2155865311054,
                    327.2214386771292,
                    327.21889294080415,
                    327.2142013147521
                ]
            },
            "time": 1.0,
            "time_step": 1
        }
    ]
``` 

and would like to have an extra dict entry for each property in `[Distributions]`.

Best wishes
Josh

**Comment by GiudGiud** [comment]
@grmnptr @somu15 

I don't see an option but it seems a good idea to me

**Comment by somu15** [comment]
You may use `SamplerData` VPP. See moose/modules/stochastic_tools/test/tests/samplers/monte_carlo/monte_carlo_weibull.i as an example.

**Reply by jvwilliams23** [reply]
Got it, thanks!

FYI, this did not add anything to the json file. But when I set `csv = true` in `[Outputs]`, I get a csv file with the sampler data and another with the `StochasticReporter` output, which I prefer.

**Reply by GiudGiud** [reply]
For others, 
I think there s an option in outputs to convert VPP output to reporter output, to use json

**Comment by somu15** [comment, marked as answer]
Yes, only reporter objects output to json. Try if https://mooseframework.inl.gov/source/reporters/StochasticMatrix.html works.
